File: 이건희/level2/오픈채팅방.py
def solution(record):
    result = []
    save_keyword = []
    save_nickname = {}
    
    for i in record:
        split_list = list(i.split(' '))
        if split_list[0] == "Enter":
            save_keyword.append([split_list[0], split_list[1]])
            save_nickname[split_list[1]] = split_list[2]
        elif split_list[0] == "Leave":
            save_keyword.append([split_list[0], split_list[1]])
        else:
            save_nickname[split_list[1]] = split_list[2]

    for i in save_keyword:
        if i[0] == "Enter":
            result.append(save_nickname[i[1]] + "님이 들어왔습니다.")
        else:
            result.append(save_nickname[i[1]] + "님이 나갔습니다.")
    
    return result

File: 이훈일/Programmers/Level2/오픈채팅방.java
import java.util.*;

class Solution {
    public String[] solution(String[] record) {
        HashMap<String, String> nickname = new HashMap<>();
        String[][] arr = new String[record.length][2];
        int count = 0;
        for(int i = 0; i < record.length; i++) {
            String[] records = record[i].split(" "); 
            
            if(records[0].equals("Change")) {
                nickname.put(records[1], records[2]);
            } else if(records[0].equals("Enter")) {
                nickname.put(records[1], records[2]);
                arr[count][0] = records[1];
                arr[count][1] = records[0];
                count++;
            } else {
                arr[count][0] = records[1];
                arr[count][1] = records[0];
                count++;
            }
        }
        
        String[] result = new String[count];
        String enter = "님이 들어왔습니다.";
        String leave = "님이 나갔습니다.";
        for(int i = 0; i < count; i++) {
            if(arr[i][1].equals("Enter")) {
                result[i] = nickname.get(arr[i][0]) + enter;
            } else if(arr[i][1].equals("Leave")) {
                result[i] = nickname.get(arr[i][0]) + leave;
            }
        }
        
        return result;
    }
}
